plugins/woocommerce-blocks/assets/js/blocks/product-template/edit.tsx

```diff
@@ -240,8 +235,29 @@ const ProductTemplateEdit = (
 			}
 			// If `inherit` is truthy, adjust conditionally the query to create a better preview.
 			if ( inherit ) {
-				if ( templateCategory ) {
-					query.categories = templateCategory[ 0 ]?.id;
+				const { taxonomy, slug } = parseTemplateSlug( templateSlug );
+
+				if ( taxonomy && slug ) {
+					const templateCategory = getEntityRecords(
+						'taxonomy',
+						taxonomy,
+						{
+							context: 'view',
+							per_page: 1,
+							_fields: [ 'id' ],
+							slug,
+						}
+					);
+
+					if ( templateCategory ) {
+						const taxonomyId = templateCategory[ 0 ]?.id;
+						if ( taxonomy === 'category' ) {
+							query.categories = taxonomyId;
+						} else {
```

Comment:
Out of curiosity, should we even handle WordPress default categories in PC block? When that could be useful? 🤔 (Probably I am missing something here)

Reply:
Hmm.. You may be right. It was there, taken from Query Loop so I didn't even think about that and didn't challenge it. 😅 

> should we even handle WordPress default categories in PC block? When that could be useful? 

While I can't think of any scenario this is useful, I also don't have confidence I can safely remove it. 🤔

Reply:
I raised that elsewhere to confirm (p1722245694194949-slack-C02FL3X7KR6). In the meantime, I suggest proceeding with this PR and if we have consensus in the related covo, I'll create a follow-up PR to remove the `category` related code. WDYT?

Reply:
Let's keep the code as per linked discussion.

Reply:
That sounds good to me! Thanks for raising this on Slack 🚀